### PR TITLE
Improve arpeggiate reading

### PR DESCRIPTION
**Pull request by SemitoneGene**
The Core::UpDown can't be used as the apreggiate reader will set the direction type to Up, which is not correct. The default arpeggiate does not have an arrow indicating up, so we need to introduce a "none" type.

### CODE REVIEW

File: Sourcecode/private/mx/impl/DirectionWriter.cpp [hunk 1]
```diff
@@ -521,4 +521,4 @@ namespace mx
             }
         }
     }
-}
+}
```
Comment (webern) [inline Sourcecode/private/mx/impl/DirectionWriter.cpp:524]: nit

```suggestion
}

```

File: Sourcecode/private/mx/impl/DirectionWriter.cpp [hunk 2]
```diff
@@ -332,33 +332,6 @@ namespace mx
...
-            }
-
-            if( myDirectionData.rehearsals.size() > 0 ) {
-                
-                for( const auto& rehearsal : myDirectionData.rehearsals ) {
-                    auto outDirType = core::makeDirectionType();
-                    outDirType->setChoice( core::DirectionType::Choice::rehearsal );
-                    this->addDirectionType( outDirType, directionPtr );
-                }
-            }
```
Comment (webern) [inline Sourcecode/private/mx/impl/DirectionWriter.cpp]: Are these code blocks no longer needed for some reason?

File: Sourcecode/private/mx/core/Enums.cpp [hunk 3]
```diff
@@ -3830,6 +3830,42 @@ namespace mx
...
+            const auto opt = tryParseUpDownNone( value );
+            return opt.value_or( UpDownNone::none );
+        }
+
+        std::optional<UpDownNone> tryParseUpDownNone( const std::string& value )
+        {
+            if( value == "up" ) { return UpDownNone::up; }
+            else if( value == "down" ) { return UpDownNone::down; }
+            return std::optional<UpDownNone>{};
+        }
```
Comment (webern) [inline Sourcecode/private/mx/core/Enums.cpp:3846]: I think you can do away with the std:optional by changing these parsing functions to return UpDownNone::none in the default case. Or is there a place where we need to know the difference between optional::none and UpDownNone::none?

Comment (SemitoneGene) [reply Sourcecode/private/mx/core/Enums.cpp:3846]: Sorry about the junk in the diff. I'll make a new one without the line ends and the sengo work that ended up in this diff.

I was afraid to dive in deeper to the use of the optional as other code uses the UpDown. The appregiate code could be smarter. As you state, it sets the default to Up, which add the direction attribute even if there was none present in the parsed XML.

**Pull request closed without merging**